### PR TITLE
ADCM-1936 Add docstrings to all methods in adcm_client

### PR DESCRIPTION
**Pull request by SofiaTal**
Add docstrings in _objects_ to all classes and methods. 
Fix pylint "consider using with" error in _stream_ and _config_data_

### CODE REVIEW

File: adcm_client/objects.py [hunk 1]
```diff
@@ -58,74 +59,88 @@ class Bundle(BaseAPIObject):
     edition = None
 
     def __repr__(self):
+        """Override base function to represent full information about bundle"""
```
Comment (nagarin) [inline adcm_client/objects.py:62]: No need to document __standard__ methods

**Review by a-alferov — DISMISSED** [review]

**Review by acmnu — CHANGES_REQUESTED** [review]

File: adcm_client/objects.py [hunk 2]
```diff
@@ -1164,19 +1278,25 @@ def upload_from_fs(self, dirname, **args) -> Bundle:
...
             result.append(self._upload(st))
         return result
 
     @allure_step('Upload bundle from {url}')
     def upload_from_url(self, url) -> Bundle:
+        """Upload bundle from {url}"""
         return self._upload(stream.web(url))
 
     def bundle_delete(self, **args):
+        """Deleting bundle object"""
```
Comment (dgusakov) [inline adcm_client/objects.py:1299]:
```suggestion
        """Delete bundle object"""
```

File: adcm_client/objects.py [hunk 3]
```diff
@@ -1164,19 +1278,25 @@ def upload_from_fs(self, dirname, **args) -> Bundle:
 
     @allure_step('Upload bundles from {dirname}')
     def upload_from_fs_all(self, dirname, **args) -> BundleList:
+        """
+        Upload multiple bundles from {dirname}
+
+        Create empty bundle list by filtering on wittingly nonexisting field
+        and value.
```
Comment (dgusakov) [inline adcm_client/objects.py:1285]: Do we need this implementation description?

File: adcm_client/objects.py [hunk 4]
```diff
@@ -237,20 +269,27 @@ class HostPrototype(Prototype):
...
 
 
 ##################################################
 #           B A S E  O B J E C T
 ##################################################
 class _BaseObject(BaseAPIObject):
+    """
+    Base class 'BaseObject' for objects 'Provider',
+    'Cluster', 'Service', 'Component', 'Host' and 'ADCM'
+    """
```
Comment (dgusakov) [inline adcm_client/objects.py:292]:
```suggestion
    """
    Base class 'BaseObject' for adcm_client objects
    """
```

**Review by nagarin — DISMISSED** [review]

File: adcm_client/objects.py [hunk 5]
```diff
@@ -61,71 +62,84 @@ def __repr__(self):
         return f"<Bundle {self.name} {self.version} {self.edition} at {id(self)}>"
 
     def provider_prototype(self) -> "ProviderPrototype":
+        """Return ProviderPrototype objects"""
```
Comment (dgusakov) [inline adcm_client/objects.py:65]:
```suggestion
        """Return ProviderPrototype object"""
```

File: adcm_client/util/stream.py [hunk 6]
```diff
@@ -21,7 +21,7 @@ def file(path, **args):
         return list(build(repopath=path, **args).values())
     else:
         stream = io.BytesIO()
-        stream.write(io.open(path, 'rb').read())
+        stream.write(io.open(path, 'rb').read())  # pylint: disable=consider-using-with
```
Comment (dgusakov) [inline adcm_client/util/stream.py:24]: This warning should be fixed and not disabled. Asc @a-alferov for help if necessary.

File: adcm_client/packer/data/config_data.py [hunk 7]
```diff
@@ -9,6 +9,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+# pylint: disable=consider-using-with
```
Comment (dgusakov) [inline adcm_client/packer/data/config_data.py:12]: This warning should be fixed and not disabled.

**Review by dgusakov — CHANGES_REQUESTED** [review]

File: adcm_client/util/stream.py (hunk 6 above)
Comment (nagarin) [reply adcm_client/util/stream.py:24]: @dgusakov, that's result of some other people activity (pylint was updated in build image) so, why it should be fixed in this PR? Your change request undermines code review principles, IMHO

Comment (dgusakov) [reply adcm_client/util/stream.py:24]: If we add mutes here then we need to create a tech dept issue to fix it in the future.

Comment (dgusakov) [reply adcm_client/util/stream.py:24]: And as far as I am concerned it is a good practice to fix minor issues not directly related to the main PR scope.

File: adcm_client/objects.py (hunk 5 above)
Comment (dgusakov) [reply adcm_client/objects.py:65]: Still relevant

File: adcm_client/packer/data/config_data.py [hunk 8]
```diff
@@ -108,15 +108,19 @@ def _from_remote_git(self, key, **kwargs):
...
-            else tarfile.open(fileobj=self.tar)
-
         def conf_files(members):
             for tarinfo in members:
                 if fnmatch.fnmatch(tarinfo.name, '*config.y*ml'):
                     yield tarinfo
 
-        confs = conf_files(tar)
+        if isinstance(self.tar, str):
+            conditions = {'name': 'self.tar'}
```
Comment (a-alferov) [inline adcm_client/packer/data/config_data.py:117]: In this case, the value should be the variable `self.tar` not a  string `'self.tar'`

**Review by dgusakov — DISMISSED** [review]

File: adcm_client/packer/data/config_data.py (hunk 7 above)
Comment (SofiaTal) [reply adcm_client/packer/data/config_data.py]: Fixed in last few commits

File: adcm_client/util/stream.py (hunk 6 above)
Comment (SofiaTal) [reply adcm_client/util/stream.py]: Fixed in last few commits

File: adcm_client/packer/data/config_data.py (hunk 8 above)
Comment (SofiaTal) [reply adcm_client/packer/data/config_data.py]: Add this changes in last commit

**Review by dgusakov — APPROVED** [review]

**Review by a-alferov — APPROVED** [review]

**Pull request merged**